src/SCRIPTS/BF/CMS/common.lua

```diff
@@ -0,0 +1,120 @@
+local _ = {
+    bitmask = {
+        firstChunk = 0x80,
+        lastChunk = 0x40,
+        batchId = 0x3F,
+        rleCharValueMask = 0x7F,
+        rleCharRepeatedMask = 0x80
+    },
+    offset = {
+        meta = 1,
+        sequence = 2,
+        data = 3
+    }
+}
+
+local function cRleDecode(buf)
+    local dest = {}
+    local rpt = false
+    local c = nil
+    for i=1, #buf do
+        if (rpt == false) then
+            c = bit32.band(buf[i], _.bitmask.rleCharValueMask)
+            if (bit32.band(buf[i], _.bitmask.rleCharRepeatedMask) > 0) then
+                rpt = true
+            else
+                dest[#dest + 1] = c
+            end
+        else
+            for j=1, buf[i] do
+                dest[#dest + 1] = c
+            end
+            rpt = false
+        end
+    end
+    return dest
+end
+
+screen = {
+    config = nil,
+    buffer = {},
+    data = {},
+    batchId = 0,
+    sequence = 0,
+    reset = function()
+        screen.buffer = {}
+        screen.data = {}
+        screen.batchId = 0
+        screen.sequence = 0
+    end,
+    draw = function()
+        if (#screen.buffer ~= nil and screen.config ~= nil and #screen.buffer > 0) then
```

Comment:
This is confusing. ``#screen.buffer`` will always not be nil as long as ``screen.buffer`` exists. This will make the script crash if ``screen.buffer`` is nil for some reason. Did you mean ``screen.buffer ~= nil`` maybe?

Reply:
Looks like a mistake.  I'll take a look.